documentation/asciidoc/computers/raspberry-pi/bcm2711-bootloader.adoc

```diff
@@ -559,6 +559,33 @@ This option may be set to 0 to block self-update without requiring the EEPROM co
 
 Default: `1`
 
+==== `secure-boot` configuration properties in `config.txt`
+The following `config.txt` properties are used to program the `secure-boot` OTP settings. These changes are irreversible and can only be programmed via `RPIBOOT` when flashing the bootloader EEPROM image.  This ensures that `secure-boot` cannot be set remotely or via accidentally inserting a stale SD card image.
+
+[[program_pubkey]]
+===== program_pubkey
+If this property is set to `1` then `recovery.bin` will write the hash of the public key in the EEPROM image to OTP.  Once set, the bootloader will reject EEPROM images signed with different RSA keys or unsigned images.
+
+Default: `0`
+
+[[revoke_devkey]]
+===== revoke_devkey
+If this property is set to `1` then `recovery.bin` will revoke the set an OTP value that causes the ROM to reject old versions of the bootloader which do not support `secure-boot`. Since the property permanently enables `secure-boot` the `program_pubkey` property must also be set (or set previously).
+
+Default: `0`
+
+[[program_rpiboot_gpio]]
+===== program_rpiboot_gpio
+Since there is no dedicated `nRPIBOOT` jumper on Pi 4B or Pi 400 an alternative GPIO must be used to select `RPIBOOT` mode by pulling the GPIO low. Only one GPIO may be selected and the available options are `2, 4, 5, 7, 8`. This property does not depend on `secure-boot` but please verify that this GPIO configuration does not conflict with any HATs which might pull the GPIO low during boot.
```

Comment:
From the descriptions in this PR, there appears to be a bit of a chicken-and-egg problem here?
"These changes are irreversible and can only be programmed via `RPIBOOT` ..."
"Since there is no dedicated `nRPIBOOT` jumper on Pi 4B or Pi 400 an alternative GPIO must be used to select `RPIBOOT` mode..."
:confused:

Reply:
Yeah, you'd have to erase the EEPROM first

Reply:
Is "erasing the EEPROM" also something that needs to be documented then? :thinking:

Reply:
added erase_eeprom